### PR TITLE
Further refactoring

### PR DESCRIPTION
**Pull request by inetic**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/463)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/authority.rs [hunk 1]
```diff
@@ -62,33 +62,36 @@ pub enum Authority {
...
+        MessageType::PostResponse(_)        => None,
+        MessageType::PutData(data)          => Some(data.name()),
+        MessageType::PutDataResponse(_)     => None,
+        MessageType::PutKey                 => None,
+        MessageType::AccountTransfer(_)     => None,
+        MessageType::PutPublicId(public_id) => Some(public_id.name()),
+        MessageType::PutPublicIdResponse(_) => None,
+        //MessageType::Refresh(_, _)          => Some(message.from_group()),
+        MessageType::Refresh(_,_)           => None,
+        MessageType::Unknown                => None,
```
Comment (dirvine) [inline src/authority.rs]: Yes I agree no wildcards here is better.

File: src/messages.rs [hunk 2]
```diff
@@ -55,6 +55,13 @@ pub struct ConnectResponse {
     pub connect_request_signature: Signature
 }
 
+/// Respond wiht Data or Error
+#[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
+pub enum DataOrError {
+Data(Data),
+Err(ResponseError),
+}
```
Comment (dirvine) [inline src/messages.rs]: Yea I am thinking this is a mistake here as it really needs `Err(RoutingError, SignedMessage)` where the message is the original message and error states clearly why it failed. I do not think we should ever send success messages (waste of b/w). Only default return is Get where if we have data we return it, Error cases for Get are something to further consider, but not for now.

File: src/messages.rs [hunk 3]
```diff
@@ -179,30 +194,38 @@ impl RoutingMessage {
...
+        //send_on_message.destination = match self.destination {
+        //    DestinationAddress::RelayToClient(_, b) => DestinationAddress::RelayToClient(destination, b),
+        //    DestinationAddress::RelayToNode(_, b)   => DestinationAddress::RelayToNode(destination, b),
+        //    DestinationAddress::Direct(_)           => DestinationAddress::Direct(destination),
+        //};
+        //send_on_message.authority = our_authority.clone();              send_on_message.authority = our_authority.clone();
+        //send_on_message
+
+        send_on_message.source      = SourceAddress::Direct(our_name);
+        send_on_message.destination = DestinationAddress::Direct(destination);
```
Comment (dirvine) [inline src/messages.rs]: I think this should be 
```send_on_message.destination = reply_destination();```

Comment (inetic) [inline src/messages.rs]: No, this is the "send_on" function, not reply :)

**Pull request merged**